src/lib/logical_query_plan/stored_table_node.cpp

```diff
@@ -93,6 +93,44 @@ bool StoredTableNode::is_column_nullable(const ColumnID column_id) const {
   return table->column_is_nullable(column_id);
 }
 
+std::shared_ptr<LQPUniqueConstraints> StoredTableNode::unique_constraints() const {
+  auto unique_constraints = std::make_shared<LQPUniqueConstraints>();
+
+  // We create unique constraints from selected table key constraints
+  const auto& table = Hyrise::get().storage_manager.get_table(table_name);
+  const auto& table_key_constraints = table->soft_key_constraints();
+
+  for (const TableKeyConstraint& table_key_constraint : table_key_constraints) {
+    // Discard constraints which involve pruned column(s)
+    const auto discard_key_constraint = [&]() {
+      for (const auto& column_id : table_key_constraint.columns()) {
+        //  Check whether constraint involves pruned column id(s).
+        if (std::find(_pruned_column_ids.cbegin(), _pruned_column_ids.cend(), column_id) != _pruned_column_ids.cend()) {
+          return true;
+        }
+      }
+      return false;
+    }();
+
+    if (!discard_key_constraint) {
```

Comment:
Guard clause

Reply:
Not sure what you mean.

Reply:
`if (discard_key_constraint) continue;` would save us one level of indentation.

Reply:
👍